tailormap-components/projects/core/src/lib/user-interface/attributelist/attributelist-table/attributelist-table.component.ts

```diff
@@ -318,6 +371,83 @@ export class AttributelistTableComponent implements AttributelistTable, OnInit,
+    this.statisticsMenu.menuData = { colName };
+    this.statisticsMenu.menu.focusFirstItem('mouse');
+    this.statisticsMenu.openMenu()
+  }
+
+  public onStatisticsMenuClick(colName: string, statisticsType: string) {
+    this.statisticParams.appLayer = this.dataSource.params.layerId;
+    this.statisticParams.column = colName;
+    this.statisticParams.type = statisticsType;
+    this.statisticParams.filter = this.dataSource.params.valueFilter;
+    const colIndex = this.layerStatisticValues.columns.findIndex(obj => obj.name === colName);
+    if (statisticsType === 'NONE') {
+      this.layerStatisticValues.columns[colIndex].statisticType = statisticsType;
+      this.layerStatisticValues.columns[colIndex].statisticValue = null;
+    } else {
+      this.statisticsService.statisticValue(this.statisticParams).subscribe((data: StatisticResponse) => {
+        if (data.success) {
+          this.layerStatisticValues.columns[colIndex].statisticType = statisticsType;
+          this.layerStatisticValues.columns[colIndex].statisticValue = data.result;
+        }
+      })
+    }
+  }
+
+  private refreshStatistics () {
+    this.layerStatisticValues.columns.forEach( col => {
+      if (col.statisticType !== 'NONE') {
+        this.onStatisticsMenuClick(col.name, col.statisticType);
+      }
+    })
+  }
+
+  private isStatisticViewable (colName: string): boolean {
+    const colIndex = this.layerStatisticValues.columns.findIndex(obj => obj.name === colName);
+    return  (this.layerStatisticValues.columns[colIndex].statisticType !== 'NONE' &&
+      this.layerStatisticValues.columns[colIndex].statisticValue &&
+      typeof (this.layerStatisticValues.columns[colIndex].statisticValue) === 'number')
+  }
+
+  public getStatisticTypeText(colName: string): string {
+    const colIndex = this.layerStatisticValues.columns.findIndex(obj => obj.name === colName);
+    let result = '';
+    if (colIndex >= 0) {
+      if (this.isStatisticViewable(colName)) {
+        result = StatisticTypeText[this.layerStatisticValues.columns[colIndex].statisticType];
+        if (result !== '') {
+          result += '=';
+        }
+      }
+    }
+    return result;
+  }
+
+  public getStatisticValue(colName: string): string {
+    const colIndex = this.layerStatisticValues.columns.findIndex(obj => obj.name === colName);
+    let result: string;
+    if (colIndex >= 0) {
+      if (this.isStatisticViewable(colName)) {
+        // Round the numbers to 0 or 2 decimals
+        // NOTE: Some columns with integer values are defined as double, so we will see 2 unexpected fractiondigits
+        if (this.layerStatisticValues.columns[colIndex].statisticType === 'COUNT' || this.getColumnType(colName) === 'integer') {
+          result = this.layerStatisticValues.columns[colIndex].statisticValue.toFixed();
+        } else {
+          result = this.layerStatisticValues.columns[colIndex].statisticValue.toFixed(2);
+        }
+      }
+    }
+    return result;
+  }
+
```

Comment:
Er lijkt hier best veel verweven te zijn, en de table component krijgt nu vrij veel verantwoordelijkheden.
Ik denk dat het een goed idee is om deze functionaliteit te encapsuleren in een apart deel (component of service), wat denk jij?

Reply:
Dat is denk ik wel een goed idee, in eerste instantie is het soms wat makkelijker om het bij elkaar te hebben om het werkend te krijgen en is het maar een paar functies. Maar uiteindelijk wordt het het steeds meer code....

Reply:
onStatisticsMenu heb ik nog laten staan vanwege de interactie met het template.
De rest heb ik in attributelist-statistic gezet.
In een separate actie zal ik e.e.a. ook nog eens voor het restant van de filtering vergelijkbaar oplossen.